lib/lotus/helpers/asset_tag_helpers.rb

```diff
@@ -40,6 +40,69 @@ def image(source, options = {})
       def asset_path(source)
         "/assets/#{source}" # To be implemented
       end
+
+      # Generates a video tag for the given arguments.
+      #
+      # @raise [ArgumentError] if the signature isn't respected
+      # @since x.x.x
+      # @api public
+      #
+      # @example Basic usage
+      #   <%= video('movie.mp4') %>
+      #     # => <video src="/assets/movie.mp4"></video>
+      #
+      # @example HTML attributes
+      #   <%= video('movie.mp4', autoplay: true, controls: true) %>
+      #     # => <video src="/assets/movie.mp4" autoplay="autoplay" controls="controls"></video>
+      #
+      # @example Fallback Content
+      #   <%=
+      #     video('movie.mp4') do
+      #       "Your browser does not support the video tag"
+      #     end
+      #   %>
+      #     # => <video src="/assets/movie.mp4">\nYour browser does not support the video tag\n</video>
+      #
+      # @example Tracks
+      #   <%=
+      #     video('movie.mp4') do
+      #       track kind: 'captions', src: view.asset_path('movie.en.vtt'), srclang: 'en', label: 'English'
+      #     end
+      #   %>
+      #     # => <video src="/assets/movie.mp4">\n<track kind="captions" src="/assets/movie.en.vtt" srclang="en" label="English">\n</video>
+      #
+      # @example Sources
+      #   <%=
+      #     video do
+      #       text "Your browser does not support the video tag"
+      #       source src: view.asset_path('movie.mp4'), type: 'video/mp4'
+      #       source src: view.asset_path('movie.ogg'), type: 'video/ogg'
+      #     end
+      #   %>
+      #     # => <video>\nYour browser does not support the video tag\n<source src="/assets/movie.mp4" type="video/mp4">\n<source src="/assets/movie.ogg" type="video/ogg">\n</video>
+      #
+      # @example Without any argument
+      #   <%= video %>
+      #     # => ArgumentError
+      #
+      # @example Without src and without block
+      #   <%= video(content: true) %>
+      #     # => ArgumentError
+      def video(src = nil, options = {}, &blk)
+        options ||= {}
+
+        if src.respond_to?(:to_hash)
+          options = src.to_hash
+        elsif src
```

Comment:
I went with an `elsif src` here rather than just `else` so that if `src` is `nil` it won't go ahead and attempt to create a path for it. This allows me to still check below if `options[:src]` exists so I can raise an ArgumentError exception.